Python/compile.c

```diff
@@ -3360,6 +3360,10 @@ compiler_subdict(struct compiler *c, expr_ty e, Py_ssize_t begin, Py_ssize_t end
         }
         for (i = begin; i < end; i++) {
             key = get_const_value((expr_ty)asdl_seq_GET(e->v.Dict.keys, i));
+            if (key == NULL) {
+                Py_DECREF(keys);
+                return 0;
```

Comment:
In this case an error is returned, but no exception is set. This looks like a bug to me. We have fixed a number of similar bugs recent times (thanks to Victor's runtime checks).

Reply:
@serhiy-storchaka Thanks!  Where would I see an example of Victor's runtime checks?  I didn't notice anything in local tests.

What exception do you think we should raise?  `MemoryError`, `RuntimeError` or something else?

Reply:
`PyErr_BadInternalCall()` is the most obvious.